lua/terrortown/menus/gamemode/roles/roleinspect.lua

```diff
@@ -0,0 +1,559 @@
+                end
+            else
+                -- derandomization is not enabled, weights are not in play
+                -- just show a list of players assigned the role
+                local layout = subroleForm:MakeIconLayout()
+
+                for k = 1, #srPlys do
+                    local ply = srPlys[k]
+                    local isHighlight = PlyIsHighlighted(ply)
+
+                    local plyIcon = vgui.Create("SimpleIconAvatar", layout)
+                    plyIcon:SetPlayer(ply)
+                    plyIcon:SetTooltip(ply:GetName())
+                    plyIcon:SetMouseInputEnabled(true)
+                    plyIcon:SetAlpha(isHighlight and 255 or 75)
+                    plyIcon:SetAvatarSize(roleIconSize)
+                    plyIcon:SetIconSize(roleIconSize, roleIconSize)
+                    plyIcon:SetSize(roleIconSize, roleIconSize)
+                end
+            end
+        end
+    end
+end
+
+local function PopulateFinalStage(stage, form, stageData)
+    local stageFullName = roleinspect.GetStageFullName(stage)
+
+    form:MakeHelp({
+        label = "help_" .. stageFullName,
+        params = {},
+    })
+
+    local finalOrderList = form:MakeIconLayout()
+
+    local finalRoles = stageData.extra.afterFinalRoles[1]
+
+    for ply, role in pairs(finalRoles) do
+        local roleData = roles.GetByIndex(role)
+
+        local entry = vgui.Create("DPiPPanelTTT2", finalOrderList)
+        entry:SetPadding(4)
+        entry:SetOuterOffset(4)
+
+        -- first added panel is the main one (player is the main icon)
+        local plyIcon = entry:Add("SimpleIconAvatar")
+        plyIcon:SetPlayer(ply)
+        plyIcon:SetTooltip(ply:GetName())
+        plyIcon:SetMouseInputEnabled(true)
+        plyIcon:SetAvatarSize(roleIconSize)
+        plyIcon:SetIconSize(roleIconSize, roleIconSize)
+        plyIcon:SetSize(roleIconSize, roleIconSize)
+
+        -- align bottom-right, preferred-axis X
+        ic = entry:Add("DRoleImageTTT2", RIGHT, BOTTOM)
+        ic:SetSize(roleIconSize * 2 / 3, roleIconSize * 2 / 3)
+        ic:SetMaterial(roleData.iconMaterial)
+        ic:SetColor(roleData.color)
+        ic:SetValue(true)
+        ic:SetMouseInputEnabled(true)
+        ic:SetTooltip(roleData.name)
+        ic:SetTooltipFixedPosition(0, roleIconSize * 2 / 3)
+    end
+end
+
+local populateStageTbl = {
+    [ROLEINSPECT_STAGE_PRESELECT] = PopulatePreselectRoleStage,
+    [ROLEINSPECT_STAGE_LAYERING] = PopulateLayeringRoleStage,
+    [ROLEINSPECT_STAGE_BASEROLES] = PopulateBaserolesStage,
+    [ROLEINSPECT_STAGE_SUBROLES] = PopulateSubrolesStage,
+    [ROLEINSPECT_STAGE_FINAL] = PopulateFinalStage,
+}
+
+local function PopulateUnhandledRoleStage(stage, form, stageData)
+    form:MakeHelp({
+        label = "help_" .. roleinspect.GetStageFullName(stage),
+    })
+
+    -- TODO: read decisions to try to create a crude approximation of the data in
+    -- the case of new stages
```

Comment:
is this still todo?

Reply:
Yes. This function is where it would be implemented. I don't think its very important though.

Reply:
I think you should either implement it, or remove it. Feel free to create an issue with a feature request that points to that line. TODOs in code tend to get lost and never be addressed